simcon_project/templates/create_assignment.html

```diff
@@ -0,0 +1,312 @@
+              op.innerHTML=stuTmp.email;
+              stu_list.appendChild(op);
+          }
+          for(var i=0;i<templates.length;i++) {
+              var tempTmp = templates[i].fields;
+              var op=document.createElement("option");
+              op.innerHTML=tempTmp.name;
+              temp_list.appendChild(op);
+          }
+          for(var i=0;i<subLabels.length;i++) {
+              var labelTmp = subLabels[i].fields;
+              var op=document.createElement("option");
+              op.innerHTML=labelTmp.label_name;
+              label_list.appendChild(op);
+          }
+      }
+      window.onload = function(){
+          init();
+      }
+
+      //add data to the table
+      function add2table(tblId,name,key,id){
+          var tBody = tblId.tBodies[0];
+          //Create tr and td elements
+          var tr = document.createElement('tr');
+          var tUsername = document.createElement('td');
+          var tKey = document.createElement('td');
+          //Add the data submitted by the form to each cell
+          tUsername.innerHTML = name;
+          tKey.innerHTML = key;
+          //Add and delete hyperlinks
+          var tDel = document.createElement('td');
+          tDel.innerHTML = '<a href="javascript:;">Delete</a>';
+          //Perform delete table row operation
+          var oA = tDel.children[0];
+          oA.onclick = function(){
+              tBody.removeChild(this.parentNode.parentNode);
+              var em=this.parentNode.parentNode.cells[1].innerHTML;
+              //student
+              if(id==1) {
+                  var index = addedStu.indexOf(em);
+                  if (index > -1) {
+                      addedStu.splice(index, 1);
+                  }
+              }
+              //template
+              else if(id==2){
+                  var name=this.parentNode.parentNode.cells[0].innerHTML;
+                  var tempKey=name+"--"+em;
+                  var index = addedTmp.indexOf(tempKey);
+                  if (index > -1) {
+                      addedTmp.splice(index, 1);
+                  }
+              }
+              //label
+              else{
+                  var index = addedLabel.indexOf(em);
+                  if (index > -1) {
+                      addedLabel.splice(index, 1);
+                  }
+              }
+          };
+          //Add cells and rows to the table
+          tr.appendChild(tUsername);
+          tr.appendChild(tKey);
+          tr.appendChild(tDel);
+          tBody.appendChild(tr);
+      }
+
+      function add_stu(tbl, stu){
+          var find=false;   //Record find the corresponding student or not
+          for(var i=0;i<students.length;i++){
+              var stuTmp=students[i];
+              if(stuTmp.email==stu){
+                  find=true;
+                  if(stuTmp.is_active!=1){
+                      window.alert("Add failed. This user is a not an active user");
```

Comment:
Adding Studet {insert student name} Failed, They are not a registered User

Reply:
*They are not an active user* instead of *They are not a registered User*

Reply:
Don't we want a check for is_registered too? Because that field gets flipped only after the student has finished registration.

Reply:
> Don't we want a check for is_registered too? Because that field gets flipped only after the student has finished registration.

I am not sure about the meaning of "registered". It seems that it only means whether the student has registered. The meaning of "is_active" seems to be whether the student has graduated.